.github/actions/check_artifact_size/action.yaml

```diff
@@ -0,0 +1,64 @@
+name: Check Artifact Size
+description: Check if the size of the newly created artifact specified by path is larger than the size of the previously uploaded artifact specified by name by the threshold.
+inputs:
+  workflow:
+    description: "Workflow to check artifact binary size for."
+    required: true
+  name:
+    description: "Name of the latest uploaded artifact."
+    required: true
+  path:
+    description: "Path to the artifact being checked."
+    required: true
+  threshold:
+    description: "Threshold for the increase in binary size of the artifact."
+    required: true
+runs:
+  using: "composite"
+  steps:
+      - name: Check Size
+        uses: actions/github-script@v6
+        with:
+          script: |
+            // Get the latest successful workflow run.
+            const workflowRuns = await github.rest.actions.listWorkflowRuns({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              workflow_id: '${{ inputs.workflow }}',
+              branch: 'main',
+              status: 'success',
+              per_page: 1
+            });
+
+            const latestRun = workflowRuns.data.workflow_runs[0].id;
+
+            // Get the artifact uploaded on the latest successful workflow run.
+            const allArtifacts = await github.rest.actions.listWorkflowRunArtifacts({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              run_id: latestRun
+            });
+
+            const matchArtifacts = allArtifacts.data.artifacts.filter((artifact) => {
+              return artifact.name == '${{ inputs.name }}';
+            });
+
+            if (matchArtifacts.length == 1) {
```

Comment:
I just found a bug in your previous change. There should be one `libcobalt.so` per configuration (arm soft/hardfp and x64). The `upload-artifact` action uses the name of the artifact (`libcobalt.so`) as the unique identifier and if multiple jobs upload the same artifact it will overwrite previous versions. That's why in the end you're only getting a single artifact.

You'll need to update the other action to save a unique (per configuration) name and use that name here to get the correct `libcobalt.so`.

In the current form all evergreen configs are checking the same (random) libcobalt.so from the previous run.